src/KafkaFlow.Abstractions/IDateTimeProvider.cs

```diff
@@ -7,8 +7,8 @@ namespace KafkaFlow
     /// </summary>
     public interface IDateTimeProvider
     {
-        /// <inheritdoc cref="DateTime.Now"/>
-        DateTime Now { get; }
+        /// <inheritdoc cref="DateTime.UtcNow"/>
+        DateTime UtcNow { get; }
```

Comment:
Since we are changing, what about moving to use `DateTimeOffset`?
 
https://learn.microsoft.com/en-us/dotnet/api/system.timeprovider.getutcnow?view=net-8.0

Reply:
It appears that the TimeProvider can only be accessed in .NET 8.